docs/guides/ngrok-for-ingress-to-services-in-a-consul-service-mesh-on-kubernetes.md

```diff
@@ -0,0 +1,312 @@
+
+Hashicups comes with `nginx` installed with intentions to the frontend and public-api services. We'll remove these and add our own intentions to allow traffic from the ngrok Ingress Controller to the frontend and public-api services.
+
+1. Remove the existing Service Intentions for the public-api and frontend services and add our own.
+```bash
+kubectl delete serviceintentions public-api -n consul
+kubectl delete serviceintentions frontend -n consul
+```
+
+1. Create Service Intention from ngrok to hashicups and the public-api
+
+```yaml
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: ServiceIntentions
+metadata:
+  name: ngrok-consul-frontend
+  namespace: consul
+spec:
+  destination:
+    name: frontend
+  sources:
+  - action: allow
+    name: ngrok-ingress-controller
+```
+
+```yaml
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: ServiceIntentions
+metadata:
+  name: ngrok-consul-api
+  namespace: consul
+spec:
+  sources:
+    - name: frontend
+      action: allow
+    - name: ngrok-ingress-controller
+      action: allow
+  destination:
+    name: public-api
+```
+
+1. Create ingress to your app
+
+Now that the ngrok Ingress Controller can communicate with the `frontend` service and `public-api` service through the Consul Service Mesh vis Service Intentions, we can create an ingress to route traffic to the app. We'll create an ingress to route traffic to the frontend service and the public-api service.
+
+:::caution Update This First!
+Update the line `host: $NGROK_DOMAIN_NAME` in the ingress object below to your ngrok domain name. For a free account, select something unique that is a subdomain of `ngrok.app`. For example, `my-unique-hashicups.ngrok.app`.
+:::
+
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-consul
+  namespace: consul
+spec:
+  ingressClassName: ngrok
+  rules:
+  # - host: $NGROK_DOMAIN_NAME
+  - host: bezek-hashicups-test.ngrok.app
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: frontend
+            port:
+              number: 3000
+      - path: /api
+        pathType: Prefix
+        backend:
+          service:
+            name: public-api
+            port:
+              number: 8080
+```
+
+This ingress object:
+- Uses the `ngrok` ingress class
+- The host is the ngrok domain name you selected that is static
+- There is a route for `/` that routes to the frontend service on port 3000
+- There is a route for `/api` that routes to the public-api service on port 8080
+
+Open your `NGROK_DOMAIN_NAME` domain in your browser and see the Hashicups application!
+
+TODO: Picture of Hashicups App
+
+
+## Add Oauth Protection to the App
+
+Now that we have the Hashicups application running, we can add oauth protection to it. We'll use the oauth module of the ngrok Ingress Controller to add oauth protection to the app. This will allow us to use Google oauth to protect the app.
```

Comment:
If useful to include here, you could point to our Google OAuth docs: https://ngrok.com/docs/cloud-edge/modules/oauth/google/